jupyter_server/services/contents/filemanager.py

```diff
@@ -119,6 +119,16 @@ def _checkpoints_class_default(self):
         deleting files really deletes them.""",
     )
 
+    always_delete_dir = Bool(
+        False,
+        config=True,
+        help="""If True, deleting non-empty directory will always be allowed.
+        WARNING this may result in files being definitely removed; e.g. on Windows
+        if the data size is too big for the trash/recycle bin they will be really
+        deleted. If False (default), non-empty directory will be send to trash only
+        if safe. And if ``delete_to_trash`` is True, they won't be deleted.""",
```

Comment:
Thanks @fcollonval - just had some wording suggestions.  The logic looks good to me.

```suggestion
        help="""If True, deleting a non-empty directory will always be allowed.
        WARNING this may result in files being permanently removed; e.g. on Windows,
        if the data size is too big for the trash/recycle bin the directory will be permanently
        deleted. If False (default), the non-empty directory will be sent to the trash only
        if safe. And if ``delete_to_trash`` is True, the directory won't be deleted.""",
```

Reply:
Thanks for reviewing the PR @kevin-bates